src/Api/Utilities/ExceptionHandlerFilterAttribute.cs

```diff
@@ -82,6 +82,11 @@ public override void OnException(ExceptionContext context)
             errorMessage = "Resource not found.";
             context.HttpContext.Response.StatusCode = 404;
         }
+        else if (exception is FeatureUnavailableException)
```

Comment:
❌ The current state of this PR has `FeatureUnavailableException` unused.

Reply:
It's still used in `CollectionsController` and `CollectionAuthorizationHandler`. We've used those as guards at the top of feature-flagged methods, e.g. 

```cs
if (flagEnabled)
{
  callnewMethod();
}

public void newMethod() {
  // just to make sure noone calls this accidentally...
  if (!flagEnabled)
  {
     throw FeatureUnavailableException();
  }
  // etc etc
}
```

That is maybe overkill and misuse of the exception. What do you think? Maybe we should just remove this use of it and rely on unit testing instead, then we don't need the exception at all.

Reply:
If your approach to set the result is working well I think it's best to use it everywhere and drop any inconsistent usages.